MacrobenchmarkSample/macrobenchmark/src/androidTest/java/com/example/macrobenchmark/NonExportedActivityBenchmark.kt

```diff
@@ -0,0 +1,103 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.macrobenchmark
+
+import android.content.Intent
+import androidx.benchmark.macro.CompilationMode
+import androidx.benchmark.macro.FrameTimingMetric
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.filters.SdkSuppress
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Direction
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.TimeUnit
+
+@LargeTest
+@SdkSuppress(minSdkVersion = 29)
+@RunWith(AndroidJUnit4::class)
+class NonExportedActivityBenchmark {
+    @get:Rule
+    val benchmarkRule = MacrobenchmarkRule()
+
+    @Test
+    fun start() {
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+        val device = UiDevice.getInstance(instrumentation)
+        benchmarkRule.measureRepeated(
+            packageName = PACKAGE_NAME,
+            metrics = listOf(FrameTimingMetric()),
+            // Try switching to different compilation modes to see the effect
+            // it has on frame timing metrics.
+            compilationMode = CompilationMode.None,
+            iterations = 3,
+            setupBlock = {
+                // Before starting to measure, navigate to the UI to be measured
+                val intent = Intent()
+                intent.action = ACTION
+                startActivityAndWait(intent)
+                val launchRecyclerActivity = device.findObject(
+                    By.res(PACKAGE_NAME, LAUNCH_RESOURCE_ID)
+                )
+                launchRecyclerActivity.click()
+                device.waitUntilActivity(RECYCLER_ACTIVITY_CLASS)
+            }
+        ) {
+            val recycler = device.findObject(
+                By.res(
+                    PACKAGE_NAME,
+                    RECYCLER_RESOURCE_ID
+                )
+            )
+            // Set gesture margin to avoid triggering gesture navigation
+            // with input events from automation.
+            recycler.setGestureMargin(device.displayWidth / 5)
+            for (i in 1..10) {
+                recycler.scroll(Direction.DOWN, 2f)
+                device.waitForIdle()
+            }
+            // Mark activity as finished
+            device.pressBack()
```

Comment:
why is this necessary?

Reply:
Its necessary if we don't use `NEW_TASK | CLEAR_TASK`. I am using it now.